independent-projects/qute/core/src/main/java/io/quarkus/qute/ValueResolvers.java

```diff
@@ -0,0 +1,198 @@
+package io.quarkus.qute;
+
+import io.quarkus.qute.Results.Result;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * Common value resolvers.
+ */
+public final class ValueResolvers {
+
+    static final String THIS = "this";
+
+    public static ValueResolver collectionResolver() {
+        return new ValueResolver() {
+
+            public boolean appliesTo(EvalContext context) {
+                return ValueResolver.matchClass(context, Collection.class);
+            }
+
+            @Override
+            public CompletionStage<Object> resolve(EvalContext context) {
+                return collectionResolveAsync(context);
+            }
+        };
+    }
+
+    public static ValueResolver thisResolver() {
+        return new ValueResolver() {
+
+            public boolean appliesTo(EvalContext context) {
+                return ValueResolver.matchClass(context, Object.class) && thisAppliesTo(context);
+            }
+
+            @Override
+            public CompletionStage<Object> resolve(EvalContext context) {
+                return CompletableFuture.completedFuture(context.getBase());
+            }
+        };
+    }
+
+    /**
+     * {@code foo.or(bar)},{@code foo or true},{@code name ?: 'elvis'}
```

Comment:
do we also support a plain ternary operator?

Reply:
Not now.